crates/sui-framework-build/src/unit_tests/build_tests.rs

```diff
@@ -22,3 +25,41 @@ fn generate_struct_layouts() {
         registry.contains_key("0000000000000000000000000000000000000002::tx_context::TxContext")
     );
 }
+
+#[test]
+fn generate_lock_file() {
+    let mut path = Path::new(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .to_path_buf();
+    path.push("sui-framework");
+    let lock_file_path = path.join("Move.lock");
+    let build_config = BuildConfig::new_for_testing();
+    let mut build_config = build_config;
+    build_config.config.lock_file = Some(lock_file_path.clone());
+    build_config.build(path).unwrap();
+
+    let mut lock_file = File::open(lock_file_path.clone()).expect("No Move.lock file");
+    let mut lock_file_contents = String::new();
+    lock_file
+        .read_to_string(&mut lock_file_contents)
+        .expect("Error reading Move.lock file");
+
+    let expected = expect![[r##"
+        # @generated by Move, please check-in and do not edit manually.
+
+        [move]
+        version = 0
+
+        dependencies = [
+          { name = "MoveStdlib" },
+        ]
+
+        [[move.package]]
+        name = "MoveStdlib"
+        source = { local = "deps/move-stdlib" }
+    "##]];
```

Comment:
Just using expect-test again because I'm not sure how to set up the snapshot, and the output is small. Let me know if/how I should use snapshot files instead!

Reply:
No strong opinions!